backend/dataall/cdkproxy/stacks/environment.py

```diff
@@ -414,84 +379,61 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
-            role=self.pivot_role,
-            handler="index.on_event",
-            code=_lambda.Code.from_asset(entry_point),
-            memory_size=1664,
-            description='This Lambda function is a cloudformation custom resource provider for LakeFormation Storage Locations '
-                        'as the Cfn resource cannot handle pivotRole updates',
-            timeout=Duration.seconds(5 * 60),
-            environment={
-                'envname': self._environment.name,
-                'LOG_LEVEL': 'DEBUG',
-                'AWS_ACCOUNT': self._environment.AwsAccountId,
-                'DEFAULT_ENV_ROLE_ARN': self._environment.EnvironmentDefaultIAMRoleArn,
-                'DEFAULT_CDK_ROLE_ARN': self._environment.CDKRoleArn,
-            },
-            dead_letter_queue_enabled=True,
-            dead_letter_queue=datalakelocation_cr_dlq,
-            on_failure=lambda_destination.SqsDestination(datalakelocation_cr_dlq),
-            tracing=_lambda.Tracing.ACTIVE,
-            runtime=_lambda.Runtime.PYTHON_3_9,
-        )
-
-        datalake_location_provider = cr.Provider(
-            self,
-            f"{self._environment.resourcePrefix}DatalakeLocationProvider",
-            on_event_handler=datalake_location_custom_resource
-        )
-
-        ssm.StringParameter(
-            self,
-            "DatalakeLocationCustomResourceFunctionArn",
-            string_value=datalake_location_custom_resource.function_arn,
-            parameter_name=f"/dataall/{self._environment.environmentUri}/cfn/custom-resources/datalocationhandler/lambda/arn",
-        )
-
-        ssm.StringParameter(
-            self,
-            "DatalakeLocationCustomResourceFunctionName",
-            string_value=datalake_location_custom_resource.function_name,
-            parameter_name=f"/dataall/{self._environment.environmentUri}/cfn/custom-resources/datalocationhandler/lambda/name",
-        )
-
-        ssm.StringParameter(
-            self,
-            'DataLocationCustomResourceProviderServiceToken',
-            string_value=datalake_location_provider.service_token,
-            parameter_name=f'/dataall/{self._environment.environmentUri}/cfn/custom-resources/datalocationhandler/provider/servicetoken',
-        )
-
         # Create SNS topics for subscriptions
         if self._environment.subscriptionsEnabled:
-            queue_key = kms.Key(
+            subscription_key_policy = iam.PolicyDocument(
+                assign_sids=True,
+                statements=[
+                    iam.PolicyStatement(
+                        actions=[
+                            "kms:Encrypt",
+                            "kms:Decrypt",
+                            "kms:ReEncrypt*",
+                            "kms:GenerateDataKey*",
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        principals=[self.default_role] + group_roles,
```

Comment:
At the moment subscriptions are a bit unused. It is just an SNS topic created for users to use. But, do you think it could happen that we send sharing or dataset update notifications from data.all backend? In that case I think the pivotRole will be added or not?

Reply:
Yes it is just an SQS Queue subscribing to an SNS Topic - it depends on what user/role would be sending the messages to the SQS Queue

I would opt to leave this as is and work on subscriptions in a separate PR adding the Pivot Role as an additional key user there if needed. At this current state, I think we are good to leave it as is to keep it as least permissive as possible but what do you think?